frontend/src/scenes/notebooks/Notebook/notebookSidebarLogic.ts

```diff
@@ -57,6 +57,20 @@ export const notebookSidebarLogic = kea<notebookSidebarLogicType>([
         ],
     })),
 
+    selectors(({ actions }) => ({
+        notebookLinkClicked: [
+            (s) => [s.notebookSideBarShown],
+            (notebookSideBarShown): ((shortId: string, internal: boolean) => void) => {
+                return (shortId: string, internal: boolean) => {
```

Comment:
ah wait... is this bad advice from me I wonder 

@Twixes calling actions in Selectors... verboten?

Reply:
It seems to be done elsewhere
https://github.com/PostHog/posthog/blob/1621ad956c6c7d3e0df8f20b6c25eef75fcbd612/frontend/src/layout/navigation-3000/sidebars/insights.ts#L45-L138
Whether that means it is right or not 🤷

Reply:
Perhaps there should be a `notebookLinkClicked` action with an associated listener that then calls the subsequent actions?

Reply:
I think in some cases that may work, but it is verboten. Selectors are part of pure functions "ecosystem" in Kea, no side effects allowed (along with actions, and reducers). For side effects, reach for [listeners](https://keajs.org/docs/core/listeners/) and [subscriptions](https://keajs.org/docs/plugins/subscriptions/)

Reply:
@daibhin So, the code you linked doesn't actually call the action _in_ the selector, it just returns a value with a _callback_ to an action. That is is völlig erlaubt, since the selector is still pure that way